sql/src/main/java/org/apache/druid/sql/calcite/rel/DruidOuterQueryRel.java

```diff
@@ -85,21 +84,6 @@ public PartialDruidQuery getPartialDruidQuery()
     return partialQuery;
   }
 
-  @Override
-  public Sequence<Object[]> runQuery()
-  {
-    // runQuery doesn't need to finalize aggregations, because the fact that runQuery is happening suggests this
-    // is the outermost query and it will actually get run as a native query. Druid's native query layer will
-    // finalize aggregations for the outermost query even if we don't explicitly ask it to.
-
-    final DruidQuery query = toDruidQuery(false);
-    if (query != null) {
```

Comment:
should this null check be moved into `NativeQueryMaker` or the `DruidRel` method or can it not be null here?

Reply:
It can't be null; that was a hold-over from an earlier time when `null` was used by the old semi-join rel to mean "known to be empty". The new join rel doesn't work that way, since it pushes computation down into the native layer.